code/segmentedtickerplant/pubsub.q

```diff
@@ -0,0 +1,110 @@
+// Pub/sub utilities for segmented tp process
+// Functionality for clients to subscribe to all tables or a subset
+// Includes option for subsrcibe to apply filters to received data
+
+\d .stpps
+
+// List of pub/sub tables, populated on startup
+t:`
+
+// Handles to publish all data
+subrequestall:enlist[`]!enlist ()
+
+// Handles and conditions to publish filtered data
+subrequestfiltered:([]tbl:`$();handle:`int$();filts:();columns:())
+
+// Function to send end of period messages to subscribers
+// Assumes that .u.endp has been defined on the client side
+endp:{
+  (neg raze union/[value subrequestall;exec handle from .stpps.subrequestfiltered])@\:(`.u.endp;x;y);
+ };
+
+// Function to send end of day messages to subscribers      
+// Assumes that .u.end has been defined on the client side   
+end:{
+  (neg raze union/[value subrequestall;exec handle from .stpps.subrequestfiltered])@\:(`.u.end;x;y);
+ };
```

Comment:
I wonder if it'd be better to put these into one function which takes one arg and does something like:
(replace 2 backticks with 1, can't get the github formatting to work)
`(neg raze union/[value subrequestall;exec handle from .stpps.subrequestfiltered])@\: $[1<count x;``.u.endp;``.u.end],x`